test/splitWith.js

```diff
@@ -0,0 +1,21 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+
+describe('splitWith', function() {
+  it('splits an array at every instance of predicate matching', function() {
+    eq(R.splitWith(R.equals(2), [1, 2, 3, 4, 2, 1]), [[1], [3, 4], [1]]);
+  });
+  it('retains all original elements', function() {
+    eq(R.splitWith(R.equals(2), [1, 1, 1]), [[1, 1, 1]]);
+  });
+  it('returns sane default if an empty list is given', function() {
+    eq(R.splitWith(R.equals(1), []), []);
+  });
+  it('handles first and last elements matching predicate', function() {
+    eq(R.splitWith(R.equals(1), [1, 2, 3, 4, 1]), [[2, 3, 4]]);
+  });
+  it('handles strings', function() {
+    eq(R.splitWith(R.equals('a'), 'bananas'), [['b'],['n'],['n'],['s']]);
+  });
+});
```

Comment:
I would want a few additional tests.  

  - Two for the issue mentioned above with two or more predicate-matching values at the ends
  - One for a non-`equals` predicate.  (`splitWith(n => n > 5, [2, 4, 6, 1, 3, 7, 8, 0])`, for instance.
  - And possibly one for an `equals` with something other than numbers or strings, perhaps `{x: 42}` or some such.

On the other hand, the `handles strings` test is not important in itself.  I would just do one or two with numbers, and some others with Strings.